### ISSUE TITLE
Only TVI(24), ARTV, Porto Canal, Euronews and Kuriakos TV work

### ISSUE THREAD

**Issue by rodajc** [marmelo/streaming-pt#51]
Rest is broken

Most give:
[ffmpeg] https: Will reconnect at 0 in 0,1,3,7 second(s), error=End of file.

SIC channels give:
Failed to open http://live.impresa.pt/live/sicnot/sicnot.m3u8.
Failed to open http://live.impresa.pt/live/siccaras/siccaras.m3u8.
Failed to open http://live.impresa.pt/live/sick/sick.m3u8.

Too bad. We'll have to watch TVI then ;-)

**Comment by marado** [comment]
Just tested...
* 1-7 working
* 8 not working
* 9-13 working
* 14 working, but black screen and no audio
* 15-19 working

@rodajc can you confirm you have streaming-pt updated and are having problems with some of these channels that are working for me?

**Comment by fimdomeio** [comment]
- 1-2  not working
- 3-4 working
- 5 not working
- 6-7 working
- 8-12 not working
- 13 working until buffer gets to 0 then appears to be dead
- 14 black screen
- 15-19  working

I believe it's the same a @marado with anything RTP also not working.

Just for reference tried streamlink and works fine with RTP channels `streamlink "https://www.rtp.pt/play/direto/rtp1" best`

**Comment by marado** [comment]
This is weird... I just tried again and have the same results as yesterday (with the exception of the caveat you found for 13, that also happens to me!).

What's your mpv version? Can you paste the output of trying to play the first stream (RTP1), for eg.?

**Comment by fimdomeio** [comment]
mpv
```
mpv 0.32.0 Copyright © 2000-2020 mpv/MPlayer/mplayer2 projects
 built on UNKNOWN
ffmpeg library versions:
   libavutil       56.31.100
   libavcodec      58.54.100
   libavformat     58.29.100
   libswscale      5.5.100
   libavfilter     7.57.100
   libswresample   3.5.100
ffmpeg version: 4.2.4-1ubuntu0.1
```
output:
```
Which TV channel do you want to watch? 1
[ffmpeg] https: Will reconnect at 0 in 0 second(s), error=End of file.
[ffmpeg] https: Will reconnect at 0 in 1 second(s), error=End of file.
[ffmpeg] https: Will reconnect at 0 in 3 second(s), error=End of file.
[ffmpeg] https: Will reconnect at 0 in 7 second(s), error=End of file.
```

Now for the interesting part.

`curl 'https://streaming-live.rtp.pt/liverepeater/smil:rtp1.smil/playlist.m3u8?pxt_rtp'` returns 204 (No Content)

`curl 'https://streaming-live.rtp.pt/liverepeater/smil:rtp1.smil/playlist.m3u8' -H 'User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OSX 10.15; rv:80.0) Gecko/20100101 Firefox/80.0' returns

```
#EXTM3U
#EXT-X-VERSION:5
#EXT-X-STREAM-INF:BANDWIDTH=1290000
chunklist_b1290000_slpor.m3u8
#EXT-X-STREAM-INF:BANDWIDTH=640000
chunklist_b640000_slpor.m3u8
```

and mpv works with:
`mpv --user-agent="SOME RANDOM USER AGENT" https://streaming-live.rtp.pt/liverepeater/smil:rtp1.smil/playlist.m3u8`

My best guess is that there is a black list for user agents. @marado mpv user-agent is not black listed mine is.

**Comment by rodajc** [comment]
$ mpv

mpv 0.32.0 Copyright © 2000-2020 mpv/MPlayer/mplayer2 projects
 built on UNKNOWN
ffmpeg library versions:
   libavutil       56.31.100 (runtime 56.51.100)
   libavcodec      58.54.100 (runtime 58.91.100)
   libavformat     58.29.100 (runtime 58.45.100)
   libswscale      5.5.100 (runtime 5.7.100)
   libavfilter     7.57.100 (runtime 7.85.100)
   libswresample   3.5.100 (runtime 3.7.100)
ffmpeg version: n4.3.1

@fimdomeio: Are you in Portugal and do you connect without proxy or anything?

I am trying to connect from abroad. In the android RTP Play app RTP1 and RTP2 don't work for that reason, but RTPi does.

**Comment by fimdomeio** [comment]
Yes, from portugal. ISP NOS (internet móvel)

**Comment by rodajc** [comment]
In that case we have different issues. I am happy with TVI anyway :-)

**Comment by marmelo** [comment]
I am also experiencing this issue using NOS.

**Comment by marmelo** [comment]
Thanks @rodajc, @marado and @fimdomeio for fixing this.
Sorry for the delay in approving the merge request.

This has been fixed by #52.
